### PR TITLE
MM-44529 - add pannel trial info in subscription page

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR adds an information banner about the trial and when it ends in the license subscription page.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44529

#### Related Pull Requests
n/a

#### Screenshots
Captura de pantalla 2022-06-06 a las 14 29 55

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/admin_console/billing/billing_subscriptions/cloud_trial_banner.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    const {formatMessage} = useIntl();
+    const openSalesLink = useOpenSalesLink();
+
+    if (trialEndDate === 0) {
+        return null;
+    }
+
+    return (
+        <AlertBanner
+            mode={'info'}
```
Comment (julmondragon) [inline components/admin_console/billing/billing_subscriptions/cloud_trial_banner.tsx:32]: Per the design it looks like this should be dismiss-able?

File: components/admin_console/billing/billing_subscriptions/cloud_trial_banner.scss [hunk 2]
```diff
@@ -0,0 +1,12 @@
+@import 'utils/variables';
+
+.upgradeLink {
```
Comment (julmondragon) [inline components/admin_console/billing/billing_subscriptions/cloud_trial_banner.scss:3]: Could reuse primary button and medium button styles from mixins.
https://github.com/mattermost/mattermost-webapp/blob/master/sass/utils/_mixins.scss#L123
https://github.com/mattermost/mattermost-webapp/blob/master/sass/utils/_mixins.scss#L206

**Review by julmondragon — APPROVED** [review]
Looks good!

File: components/admin_console/billing/billing_subscriptions/cloud_trial_banner.tsx (hunk 1 above)
Comment (pvev) [reply components/admin_console/billing/billing_subscriptions/cloud_trial_banner.tsx:32]: you are right @julmondragon, thanks!. @marianunez @annelieseklein once the admin dismisses this banner, should it be dismissed just for the session or should be dismissed for the remaining duration of the trial (meaning dismissed for good)?

Comment (annelieseklein) [reply components/admin_console/billing/billing_subscriptions/cloud_trial_banner.tsx:32]: @johndavidlugtu and I are reconvening on this to make a final decision, will update shortly

Comment (annelieseklein) [reply components/admin_console/billing/billing_subscriptions/cloud_trial_banner.tsx:32]: @pvev it should be dismissed/snoozed for 10 days just for that admin

File: components/admin_console/billing/billing_subscriptions/index.tsx [hunk 3]
```diff
@@ -154,6 +156,7 @@ const BillingSubscriptions = () => {
                     {showCreditCardBanner &&
                         isCardExpired &&
                         creditCardExpiredBanner(setShowCreditCardBanner)}
+                    {(isCloudFreeEnabled && isFreeTrial) && (<CloudTrialBanner trialEndDate={trialEndDate}/>)}
```
Comment (julmondragon) [inline components/admin_console/billing/billing_subscriptions/index.tsx:159]: Should this check also for cloud license?

**Review by julmondragon — APPROVED** [review]
LGTM!

File: components/admin_console/billing/billing_subscriptions/index.tsx (hunk 3 above)
Comment (pvev) [reply components/admin_console/billing/billing_subscriptions/index.tsx:159]: no need, this is in subscription page that is only shown when in cloud.

**Review by stevemudie — APPROVED** [review]
@pvev Thanks for the video and explanation. It looks good here and I'll of course test it again post merge.

The E2E came through great so this is all set.

**Review by JulienTant — APPROVED** [review]
Solid @pvev! 🚀

**Review by marianunez — COMMENTED** [review]
@pvev took a look at the test server and banner is looking good. However, it seems we are still missing the change to include the days left in the `trial` tag as shown in the design (second bullet in the ticket) 

https://www.figma.com/file/Aob5ffHtfjKzQoebR93ycA/Cloud-Freemium?node-id=657%3A431452

**Comment by pvev** [comment]
> @pvev took a look at the test server and banner is looking good! However, it seems we are still missing the change to include the days left in the `trial` tag as shown in the design (second bullet in the ticket)
> 
> https://www.figma.com/file/Aob5ffHtfjKzQoebR93ycA/Cloud-Freemium?node-id=657%3A431452
done @marianunez 
Captura de pantalla 2022-06-09 a las 14 39 41

**Review by marianunez — APPROVED** [review]
Thanks @pvev!

**Pull request merged**